graphrag/index/context_switch/contextSwitcher.py

```diff
@@ -1,22 +1,247 @@
+)
+from common.graph_db_client import GraphDBClient
+import os
+from graphrag.vector_stores import VectorStoreFactory, VectorStoreType
+from graphrag.vector_stores.base import BaseVectorStore
+from graphrag.vector_stores.lancedb import LanceDBVectorStore
+from graphrag.vector_stores.kusto import KustoVectorStore
+from graphrag.query.indexer_adapters import (
+    read_indexer_covariates,
+    read_indexer_entities,
+    read_indexer_relationships,
+    read_indexer_reports,
+    kt_read_indexer_reports,
+    read_indexer_text_units,
+)
+from graphrag.model.entity import Entity
 
 class ContextSwitcher:
     """ContextSwitcher class definition."""
 
-    def __init__(self):
-        #initialize Gremline and Cosmos Db client here.
-        pass
-    def activate(self, config: GraphRagConfig | str, contextId: str | None, reporter: ProgressReporter):
+    def __init__(self, root_dir:str , config_dir:str,reporter: ProgressReporter,
+                 context_id:str, community_level:int ,
+                 data_dir: str = None,
+                 optimized_search: bool= False):
+
+        self.root_dir=root_dir
+        self.config_dir=config_dir
+        self.data_dir=data_dir
+        self.reporter=reporter
+        self.context_id=context_id
+        self.optimized_search=optimized_search
+        self.community_level = community_level
+
+    def set_ctx_activation(
+            self,
+            activate: int,
+            entities: list[Entity]=[],
+            config_args: dict | None = None,
+        ):
+        if not config_args:
+                config_args = {}
+
+        collection_name = config_args.get(
+                "query_collection_name", "entity_description_embeddings"
+        )
+
+        collection_name += "_" + self.context_id
+        config_args.update({"collection_name": collection_name})
+        vector_name = config_args.get(
+                "vector_search_column", "description_embedding"
+        )
+        config_args.update({"vector_name": vector_name})
+
+        description_embedding_store = VectorStoreFactory.get_vector_store(
+                vector_store_type=VectorStoreType.Kusto, kwargs=config_args
+        )
+        description_embedding_store.connect(**config_args)
+
+        if activate:
+            description_embedding_store.load_entities(entities)
+        else:
+            description_embedding_store.unload_entities()
+
+        return 0
+
+    def activate(self):
         """Activate the context."""
         #1. read the context id to fileId mapping.
         #2. read the file from storage using common/blob_storage_client.py
         #3. GraphDB: use cosmos db client to load data into Cosmos DB.
         #4. KustoDB: use Kusto client to load embedding data into Kusto.
 
-        return 0
+        data_dir=self.data_dir
+        root_dir=self.root_dir
+        config_dir=self.config_dir
+        reporter=self.reporter
+        context_id=self.context_id
+        optimized_search=self.optimized_search
```

Comment:
we should get everything out related to kusto in separate module.
Something like below.
Activate:
GraphDb.Provision(contextId
KustoDb.Provision(ContextId)
For each path in File_paths:
      KustoDb.Ingest(path)
     GraphDb.Ingest(Path)